modules/schema/src/main/scala/lucuma/odb/json/sourceprofile.scala

```diff
@@ -88,6 +92,61 @@ trait SourceProfileCodec {
        .mapObject(_.add("fwhm", g.fwhm.asJson))
     }
 
+  private def refinedBigDecimalCodec[V](
+    name:           String,
+    range:          String,
+    toBigDecimal:   V => BigDecimal,
+    fromBigDecimal: BigDecimal => Either[String, V]
+  ): Codec[V] =
+    new Codec[V] {
+      def apply(v: V): Json =
+        toBigDecimal(v).asJson
+
+      def apply(c: HCursor): Decoder.Result[V] =
+        c.as[BigDecimal].flatMap { bd =>
+          fromBigDecimal(bd).leftMap(_ => DecodingFailure(s"Illegal $name value, $bd. Must be in $range", c.history))
+        }
+    }
+
+  given Codec[BrightnessValue] =
+    refinedBigDecimalCodec[BrightnessValue](
+      "brightness value",
+      "[-30, 100,000,000]",
```

Comment:
I'm a bit worried of having the limits hardcoded here. If, for some reason, we change them in core, we have to remember to change them here too.

There might be a way to pull them from the refined type? Otherwise, we can just make them available in core.

Reply:
I thought the same thing but I'm not sure how to do it.